orx-compositor/src/commonMain/kotlin/Compositor.kt

```diff
@@ -165,27 +169,56 @@ open class Layer internal constructor() {
+                        drawer.image(resolvedColorBuffer!!, layerPost.bounds, drawer.bounds)
+                    }
                 }
             } else {
                 localBlendFilter.first.apply(localBlendFilter.second)
-                localBlendFilter.first.apply(arrayOf(activeRenderTarget.colorBuffer(0), layerPost), activeRenderTarget.colorBuffer(0))
-            }
+                activeRenderTarget.colorBuffer(0).copyTo(activeRenderTargetColorBuffer!!)
+                if (bufferMultisample == BufferMultisample.Disabled) {
+                    localBlendFilter.first.apply(arrayOf(activeRenderTargetColorBuffer!!, layerPost),
+                        activeRenderTargetColorBuffer!!)
+                } else {
+                    layerPost.copyTo(resolvedColorBuffer!!)
+                    localBlendFilter.first.apply(arrayOf(activeRenderTargetColorBuffer!!, resolvedColorBuffer!!),
+                        activeRenderTargetColorBuffer!!)
+                }
 
-            accumulation?.copyTo(target.colorBuffer(0))
+                if (activeRenderTarget !is ProgramRenderTarget) {
+                    activeRenderTargetColorBuffer!!.copyTo(target.colorBuffer(0))
+                }
+                activeRenderTargetColorBuffer!!.copyTo(activeRenderTarget.colorBuffer(0))
+            }
         }
     }
 
     private fun shouldCreateLayerTarget(activeRenderTarget: RenderTarget): Boolean {
-        return layerTarget == null || ((layerTarget?.width != activeRenderTarget.width || layerTarget?.height != activeRenderTarget.height) && activeRenderTarget.width > 0 && activeRenderTarget.height > 0)
+        return layerTarget == null
+               || ((layerTarget?.width != activeRenderTarget.width || layerTarget?.height != activeRenderTarget.height)
+                   && activeRenderTarget.width > 0 && activeRenderTarget.height > 0)
     }
 
-    private fun createLayerTarget(activeRenderTarget: RenderTarget, drawer: Drawer) {
+    private fun createLayerTarget(
+        activeRenderTarget: RenderTarget, drawer: Drawer, bufferMultisample: BufferMultisample
+    ) {
         layerTarget?.deepDestroy()
-        layerTarget = renderTarget(activeRenderTarget.width, activeRenderTarget.height) {
+        layerTarget = renderTarget(activeRenderTarget.width, activeRenderTarget.height,
+            activeRenderTarget.contentScale, bufferMultisample) {
             colorBuffer(type = colorType)
             depthBuffer()
         }
+        if (bufferMultisample != BufferMultisample.Disabled) {
+            resolvedColorBuffer?.destroy()
+            resolvedColorBuffer = colorBuffer(activeRenderTarget.width, activeRenderTarget.height,
+                activeRenderTarget.contentScale, type = colorType)
+        }
+        activeRenderTargetColorBuffer?.destroy()
+        activeRenderTargetColorBuffer = colorBuffer(activeRenderTarget.width, activeRenderTarget.height,
+            activeRenderTarget.contentScale, type = colorType)
```

Comment:
I don't actually know what "active" rendertarget means or if there's any semantics associated with it. I'm assuming it can change dimensions so I'm recreating it here if it does so.